src/Aspire.Cli/Commands/PublishCommand.cs

```diff
@@ -301,4 +219,68 @@ protected override async Task<int> ExecuteAsync(ParseResult parseResult, Cancell
         }
     }
+
+    public static async Task<bool> ProcessPublishingActivitiesAsync(IAsyncEnumerable<(string Id, string StatusText, bool IsComplete, bool IsError)> publishingActivities, CancellationToken cancellationToken)
+    {
+        var lastActivityUpdateLookup = new Dictionary<string, (string Id, string StatusText, bool IsComplete, bool IsError)>();
+        await foreach (var publishingActivity in publishingActivities.WithCancellation(cancellationToken))
+        {
+            lastActivityUpdateLookup[publishingActivity.Id] = publishingActivity;
+
+            if (lastActivityUpdateLookup.Any(kvp => kvp.Value.IsError) || lastActivityUpdateLookup.All(kvp => kvp.Value.IsComplete))
+            {
+                // If we have an error or all tasks are complete then we can stop
+                // processing the publishing activities.
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public static async Task<bool> ProcessAndDisplayPublishingActivitiesAsync(IAsyncEnumerable<(string Id, string StatusText, bool IsComplete, bool IsError)> publishingActivities, CancellationToken cancellationToken)
+    {
+        return await AnsiConsole.Progress()
+            .AutoRefresh(true)
+            .Columns(
+                new TaskDescriptionColumn() { Alignment = Justify.Left },
+                new ProgressBarColumn() { Width = 10 },
+                new ElapsedTimeColumn())
+            .StartAsync<bool>(async context => {
+
+                var progressTasks = new Dictionary<string, ProgressTask>();
+
+                await foreach (var publishingActivity in publishingActivities.WithCancellation(cancellationToken))
+                {
+                    if (!progressTasks.TryGetValue(publishingActivity.Id, out var progressTask))
+                    {
+                        progressTask = context.AddTask(publishingActivity.Id);
+                        progressTask.StartTask();
+                        progressTask.IsIndeterminate();
+                        progressTasks.Add(publishingActivity.Id, progressTask);
+                    }
+
+                    progressTask.Description = $":play_button:  {publishingActivity.StatusText}";
+
+                    if (publishingActivity.IsComplete && !publishingActivity.IsError)
+                    {
+                        progressTask.Description = $":check_mark:  {publishingActivity.StatusText}";
+                        progressTask.Value = 100;
+                        progressTask.StopTask();
+                    }
+                    else if (publishingActivity.IsError)
+                    {
+                        progressTask.Description = $"[red bold]:cross_mark:  {publishingActivity.StatusText}[/]";
+                        progressTask.Value = 0;
+                        return false;
+                    }
+                    else
+                    {
+                        // Keep going man!
+                    }
```

Comment:
[nit] Do we need this else clause at all? 

[minor nit] Consider rewording the comment to something more neutral.